### ISSUE TITLE
Bug: Update class span start when using decorators and export

### ISSUE THREAD

**Issue by ArnaudBarre** [typescript-eslint/typescript-eslint#10929]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&showAST=es&fileType=.ts&code=AIIwhgTgUApgHgBwPYQC4AIDGAbMBnPdAMSSXQG8BfIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
@bar
export class Foo {}
```

### Expected Result

The `start` location for the class should be `0` to match Babel behaviour.
Tagging @nicolo-ribaudo who worked on it

### Actual Result

ClassDeclaration.start = 12

### Additional Info

Playground for Babel: https://ast-explorer.dev/#eNolijEOwyAMRa+CPOcEmTr1BN3qDoagKhXFyIaqEeLuMcr2/vuvQ4IVPvQjDbKXCgsUE558TMbB+OZJMMd/YakuJFJ1d2bXB2Yr2IqO2TkE5SYhPo4SEVbbX95aMl6uu6T23rPO7znNdFsMLFRZTE/3wjxgnBJ4MEA=

### Versions

`@typescript-eslint/parser`: `8.26.0`

**Comment by bradzacher** [comment]
> should be 0 to match Babel behaviour

Sorry, could you clarify - why should we match the babel behaviour? Why is their behaviour more correct than ours?

**Comment by ArnaudBarre** [comment]
Actually with the ESTree plugin `ExportNamedDeclaration.start` is also 5 in Babel.
Ok I found this discussion https://github.com/estree/estree/issues/315 that is my issue: what should tool like OXC generate as span.
